src/content/platform-integration/macos/platform-views.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Hosting native macOS views in your Flutter app with Platform Views
+short-title: macOS platform-views
+description: Learn how to host native macOS views in your Flutter app with Platform Views.
+---
+
+<?code-excerpt path-base="platform_integration/platform_views"?>
+
+Platform views allow you to embed native views in a Flutter app, so you can
+apply transforms, clips, and opacity to the native view from Dart.
+
+This allows you, for example, to use the native web views directly inside your
+Flutter app.
+
+:::note
+This page discusses how to host your own native macOS views within a Flutter
+app. If you'd like to embed native iOS views in your Flutter app, see
+[Hosting native iOS views][].
+:::
+
+[Hosting native iOS views]: /platform-integration/ios/platform-views
+
+macOS uses Hybrid composition, which means that the native `NSView` is appended
+to the view hierarchy.
+
+To create a platform view on macOS, use the following instructions:
+
+## On the Dart side
+
+On the Dart side, create a `Widget` and add the build implementation, as shown
+in the following steps.
+
+In the Dart widget file, make changes similar to those 
+shown in `native_view_example.dart`:
+
+<ol>
+<li>
+
+Add the following imports:
+
+<?code-excerpt "lib/native_view_example_4.dart (import)"?>
+```dart
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+```
+
+</li>
+
+<li>
+
+Implement a `build()` method:
+
+<?code-excerpt "lib/native_view_example_4.dart (macos-composition)"?>
+```dart
+Widget build(BuildContext context) {
+  // This is used in the platform side to register the view.
+  const String viewType = '<platform-view-type>';
+  // Pass parameters to the platform side.
+  final Map<String, dynamic> creationParams = <String, dynamic>{};
+
+  return AppKitView(
+    viewType: viewType,
+    layoutDirection: TextDirection.ltr,
+    creationParams: creationParams,
+    creationParamsCodec: const StandardMessageCodec(),
+  );
+}
+```
+
+</li>
+</ol>
+
+For more information, see the API docs for: [`AppKitView`][].
+
+[`AppKitView`]: {{site.api}}/flutter/widgets/AppKitView-class.html
+
+## On the platform side
+
+On the platform side, use Swift:
+
+{% samplecode "macos-platform-views", "Swift" %}
+{% sample "Swift" %}
+
+Implement the factory and the platform view. The `FLNativeViewFactory` creates
+the platform view, and the platform view provides a reference to the `NSView`.
+For example, `FLNativeView.swift`:
+
+```swift
+import Cocoa
+import FlutterMacOS
+
+class FLNativeViewFactory: NSObject, FlutterPlatformViewFactory {
```

Comment:
Prefixing is an Obj-C-ism; I would remove the `FL`s from these classes.

Reply:
Done.